src/writer.rs

```diff
@@ -297,6 +336,40 @@ impl<'a> WriterOptions<'a> {
     }
 }
 
+fn write_extensions<W, E>(
```

Comment:
Why not just merge these two into one function that takes an `Option<ExtensionSequence>`?  Then you can reduce the number of calls everywhere else by removing the `write_header_terminator` call.

Reply:
Done with `Option<&ExtensionSequence>`